webapi/src/main/scala/org/knora/webapi/messages/v1/responder/permissionmessages/PermissionMessagesV1.scala

```diff
@@ -0,0 +1,801 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.knora.webapi.messages.v1.responder.permissionmessages
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import org.apache.jena.sparql.function.library.leviathan.log
+import org.knora.webapi._
+import org.knora.webapi.messages.v1.responder.permissionmessages.PermissionDataType.PermissionProfileType
+import org.knora.webapi.messages.v1.responder.projectmessages.{ProjectInfoV1, ProjectV1JsonProtocol}
+import org.knora.webapi.messages.v1.responder.usermessages.UserProfileV1
+import org.knora.webapi.messages.v1.responder.{KnoraRequestV1, KnoraResponseV1}
+import org.knora.webapi.util.MessageUtil
+import spray.json._
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Messages
+
+/**
+  * An abstract trait representing message that can be sent to `PermissionsResponderV1`.
+  */
+sealed trait PermissionsResponderRequestV1 extends KnoraRequestV1
+
+/**
+  * A message that requests the user's [[PermissionDataV1]].
+ *
+  * @param projectIris the projects the user is part of.
+  * @param groupIris the groups the user is member of.
+  * @param isInProjectAdminGroups the projects for which the user is member of the ProjectAdmin group.
+  * @param isInSystemAdminGroup the flag denoting users membership in the SystemAdmin group.
+  */
+case class PermissionDataGetV1(projectIris: Seq[IRI],
+                               groupIris: Seq[IRI],
+                               isInProjectAdminGroups: Seq[IRI],
+                               isInSystemAdminGroup: Boolean
+                                 ) extends PermissionsResponderRequestV1
+
+/**
+  * A message that requests the creation of permissions (administrative and default) for a certain project
+  * based on a predefined template. These permissions can be applied to a newly created or an exesting project.
+  * In the case of an existing project, this operation behaves destructive, in the sense that all existing permissions
+  * attached to a project are deleted, before any new permissions are created.
+  *
+  * @param projectIri the IRI of the project.
+  * @param permissionsTemplate the permissions template.
+  */
+//case class TemplatePermissionsCreateRequestV1(projectIri: IRI, permissionsTemplate: PermissionsTemplate, userProfileV1: UserProfileV1) extends PermissionsResponderRequestV1
```

Comment:
Not used at the moment, but will be used in the future. I commented it out because it caused some compile errors, and I didn't want to spend time debugging it :-)